demos/mqtt/mqtt_demo_basic_tls/mqtt_demo_basic_tls.c

```diff
@@ -625,6 +655,95 @@ static void handleIncomingPublish( MQTTPublishInfo_t * pPublishInfo,
 
 /*-----------------------------------------------------------*/
 
+static void updateSubscribeStatus( MQTTPacketInfo_t * pPacketInfo )
+{
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    uint8_t subscriptionStatus = 0;
+
+    assert( pPacketInfo != NULL );
+    assert( pPacketInfo->pRemainingData != NULL );
+
+    uint8_t * pPayload = pPacketInfo->pRemainingData;
+    size_t pSize = pPacketInfo->remainingLength;
+
+    mqttStatus = MQTT_GetSubAckStatusCodes( pPacketInfo, &pPayload, &pSize );
+
+    if( mqttStatus == MQTTSuccess )
+    {
+        /* Demo only subscribes to one topic, so only one status code is returned. */
+        subscriptionStatus = pPayload[ 0 ];
+
+        globalSubscribeStatus = subscriptionStatus;
+    }
+    else
+    {
+        globalSubscribeStatus = MQTTSubAckFailure;
+    }
+}
+
+/*-----------------------------------------------------------*/
+
+static int handleResubscribe( MQTTContext_t * pMqttContext )
+{
+    int returnStatus = EXIT_SUCCESS;
+    MQTTStatus_t mqttStatus = MQTTSuccess;
+    bool retriesArePending = true;
+    TransportReconnectParams_t reconnectParams;
+
+    assert( pMqttContext != NULL );
+
+    /* Initialize retry attempts and interval */
+    Transport_ReconnectParamsReset( &reconnectParams );
```

Comment:
Plan on opening a separate PR to rename `transport_reconnect` files/functions to a more generic "retry" pattern, since they are not specific to transport reconnection.